src/auth/src/credentials/user_credential.rs

```diff
@@ -247,7 +260,9 @@ mod test {
             let mut json = json_full.clone();
             // Remove a required field from the JSON
             json[required_field].take();
-            UserTokenProvider::from_json(json).err().unwrap();
+            serde_json::from_value::<AuthorizedUser>(json)
```

Comment:
maybe

```rust
  assert!(serde_json::from_value::<AuthorizedUser>(json).is_err(), "{json:?}");
```

is marginally more readable, it tells me you are asserting something vs. indirectly asserting something because the error did not fail.